{"name": "数据库校验-策略开仓-持仓检查", "status": "broken", "statusDetails": {"message": "pymysql.err.ProgrammingError: (1064, \"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'symbol LIKE '%XAUUSD%' AND send_no = '6096165272004' AND type = '0' AND trader_i' at line 1\")", "trace": "self = <lingkuan_youhua11.test_youhua.test_vps_ordersend_youhua.TestVPSOrderSend object at 0x0000019BB8C29790>\nvar_manager = <lingkuan_youhua11.commons.variable_manager.VariableManager object at 0x0000019BB8C3E220>\ndb_transaction = <pymysql.connections.Connection object at 0x0000019BB8CC1190>\n\n    @allure.title(\"数据库校验-策略开仓-持仓检查\")\n    def test_dbquery_order_detail(self, var_manager, db_transaction):\n        with allure.step(\"1. 根据下单指令获取跟单账号订单数据\"):\n            order_no = var_manager.get_variable(\"order_no\")\n            vps_trader_id = var_manager.get_variable(\"vps_trader_id\")\n            trader_ordersend = var_manager.get_variable(\"trader_ordersend\")\n    \n            # 使用正确格式的条件字典\n            conditions = {\n                \"like\": {\"field\": \"symbol\", \"value\": trader_ordersend[\"symbol\"]},\n                \"send_no\": order_no,\n                \"type\": trader_ordersend[\"type\"],\n                \"trader_id\": vps_trader_id\n            }\n    \n            # 使用基类的wait_for_db_record方法\n>           db_data = self.wait_for_db_record(\n                db_transaction,\n                trader_ordersend[\"table_detail\"],\n                conditions,\n                time_field=\"create_time\",\n                time_range=MYSQL_TIME,\n                timeout=WAIT_TIMEOUT,\n                poll_interval=POLL_INTERVAL\n            )\n\ntest_youhua\\test_vps_ordersend_youhua.py:179: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ncommons\\api_base.py:197: in wait_for_db_record\n    return wait_for_condition(\ncommons\\wait_utils.py:39: in wait_for_condition\n    last_result = condition()\ncommons\\api_base.py:189: in check_db\n    cursor.execute(sql, params)\n..\\.venv\\lib\\site-packages\\pymysql\\cursors.py:153: in execute\n    result = self._query(query)\n..\\.venv\\lib\\site-packages\\pymysql\\cursors.py:322: in _query\n    conn.query(q)\n..\\.venv\\lib\\site-packages\\pymysql\\connections.py:563: in query\n    self._affected_rows = self._read_query_result(unbuffered=unbuffered)\n..\\.venv\\lib\\site-packages\\pymysql\\connections.py:825: in _read_query_result\n    result.read()\n..\\.venv\\lib\\site-packages\\pymysql\\connections.py:1199: in read\n    first_packet = self.connection._read_packet()\n..\\.venv\\lib\\site-packages\\pymysql\\connections.py:775: in _read_packet\n    packet.raise_for_error()\n..\\.venv\\lib\\site-packages\\pymysql\\protocol.py:219: in raise_for_error\n    err.raise_mysql_exception(self._data)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\ndata = b\"\\xff(\\x04#42000You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version ... right syntax to use near 'symbol LIKE '%XAUUSD%' AND send_no = '6096165272004' AND type = '0' AND trader_i' at line 1\"\n\n    def raise_mysql_exception(data):\n        errno = struct.unpack(\"<h\", data[1:3])[0]\n        # https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_err_packet.html\n        # Error packet has optional sqlstate that is 5 bytes and starts with '#'.\n        if data[3] == 0x23:  # '#'\n            # sqlstate = data[4:9].decode()\n            # TODO: Append (sqlstate) in the error message. This will be come in next minor release.\n            errval = data[9:].decode(\"utf-8\", \"replace\")\n        else:\n            errval = data[3:].decode(\"utf-8\", \"replace\")\n        errorclass = error_map.get(errno)\n        if errorclass is None:\n            errorclass = InternalError if errno < 1000 else OperationalError\n>       raise errorclass(errno, errval)\nE       pymysql.err.ProgrammingError: (1064, \"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'symbol LIKE '%XAUUSD%' AND send_no = '6096165272004' AND type = '0' AND trader_i' at line 1\")\n\n..\\.venv\\lib\\site-packages\\pymysql\\err.py:150: ProgrammingError"}, "steps": [{"name": "1. 根据下单指令获取跟单账号订单数据", "status": "broken", "statusDetails": {"message": "pymysql.err.ProgrammingError: (1064, \"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'symbol LIKE '%XAUUSD%' AND send_no = '6096165272004' AND type = '0' AND trader_i' at line 1\")\n", "trace": "  File \"D:\\pycharm_test\\lingkuan_youhua11\\test_youhua\\test_vps_ordersend_youhua.py\", line 179, in test_dbquery_order_detail\n    db_data = self.wait_for_db_record(\n  File \"D:\\pycharm_test\\lingkuan_youhua11\\commons\\api_base.py\", line 197, in wait_for_db_record\n    return wait_for_condition(\n  File \"D:\\pycharm_test\\lingkuan_youhua11\\commons\\wait_utils.py\", line 39, in wait_for_condition\n    last_result = condition()\n  File \"D:\\pycharm_test\\lingkuan_youhua11\\commons\\api_base.py\", line 189, in check_db\n    cursor.execute(sql, params)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\cursors.py\", line 153, in execute\n    result = self._query(query)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\cursors.py\", line 322, in _query\n    conn.query(q)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 563, in query\n    self._affected_rows = self._read_query_result(unbuffered=unbuffered)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 825, in _read_query_result\n    result.read()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 1199, in read\n    first_packet = self.connection._read_packet()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 775, in _read_packet\n    packet.raise_for_error()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\protocol.py\", line 219, in raise_for_error\n    err.raise_mysql_exception(self._data)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\err.py\", line 150, in raise_mysql_exception\n    raise errorclass(errno, errval)\n"}, "steps": [{"name": "等待数据出现在 follow_order_detail 表中", "status": "broken", "statusDetails": {"message": "pymysql.err.ProgrammingError: (1064, \"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'symbol LIKE '%XAUUSD%' AND send_no = '6096165272004' AND type = '0' AND trader_i' at line 1\")\n", "trace": "  File \"D:\\pycharm_test\\lingkuan_youhua11\\commons\\wait_utils.py\", line 39, in wait_for_condition\n    last_result = condition()\n  File \"D:\\pycharm_test\\lingkuan_youhua11\\commons\\api_base.py\", line 189, in check_db\n    cursor.execute(sql, params)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\cursors.py\", line 153, in execute\n    result = self._query(query)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\cursors.py\", line 322, in _query\n    conn.query(q)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 563, in query\n    self._affected_rows = self._read_query_result(unbuffered=unbuffered)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 825, in _read_query_result\n    result.read()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 1199, in read\n    first_packet = self.connection._read_packet()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\connections.py\", line 775, in _read_packet\n    packet.raise_for_error()\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\protocol.py\", line 219, in raise_for_error\n    err.raise_mysql_exception(self._data)\n  File \"D:\\pycharm_test\\.venv\\lib\\site-packages\\pymysql\\err.py\", line 150, in raise_mysql_exception\n    raise errorclass(errno, errval)\n"}, "attachments": [{"name": "异常信息", "source": "0ba6d59b-8836-47f9-ab02-0b45997a8300-attachment.txt", "type": "text/plain"}], "start": 1751029006340, "stop": 1751029006387}], "start": 1751029006340, "stop": 1751029006388}], "attachments": [{"name": "log", "source": "8daa93b2-8ed6-4f50-816e-b98d799bd2e6-attachment.txt", "type": "text/plain"}], "start": 1751029006340, "stop": 1751029006388, "uuid": "5780cf02-1e42-4898-9a67-60c1e91baada", "historyId": "cd1010efbc34db8f800784b71bfcff3c", "testCaseId": "cd1010efbc34db8f800784b71bfcff3c", "fullName": "test_youhua.test_vps_ordersend_youhua.TestVPSOrderSend#test_dbquery_order_detail", "labels": [{"name": "feature", "value": "VPS策略下单-正常开仓平仓"}, {"name": "parentSuite", "value": "test_youhua"}, {"name": "suite", "value": "test_vps_ordersend_youhua"}, {"name": "subSuite", "value": "TestVPSOrderSend"}, {"name": "host", "value": "DESKTOP-4S9CU1E"}, {"name": "thread", "value": "20212-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_youhua.test_vps_ordersend_youhua"}]}